Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -58,13 +58,24 @@ bool LogFileReader::readAll(const QString& fileName)
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
+    if(sections.size() != COLUMNS)
+        return false;
 
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    bool voltageOk;
+    bool currentOk;
+
+    batteryData.voltage = sections.at(1).toDouble(&voltageOk);
+
+    batteryData.current = sections.at(2).toDouble(&currentOk);
+
+    if(!batteryData.time.isValid() || !voltageOk || !currentOk)
+    {
+        return false;
+    }
```

Comment:
You can also just `return  (batteryData.time.isValid() && voltageOk && currentOk)`